### PR TITLE
Fix edit start date bug for makeup lesson

### LINKED ISSUE #218: Add end date tests for Lesson Add and Edit logic classes

### LINKED ISSUE #249: [PE-D] Editing start time for makeup lessons throws an error
https://github.com/pattan9955/ped/issues/7#issue-1039336186

### LINKED ISSUE #254: [PE-D] Lesson conflict details not shown
When trying to add a lesson that has conflict with another existing lesson, the feedback is "This lesson clashes with an existing lesson." 

Would be great if the exact clash is specified in case the user doesn't know

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: ganhongyao/ped#7

### PR DESCRIPTION
**Pull request by eeliana**
fix #249 
fix #254 
fix #218

### CODE REVIEW

**Comment by lingshanng** [comment]
would be good to add tests in LessonEditCommand also

**Review by angkoonhwee — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/LessonEditCommand.java [hunk 1]
```diff
@@ -143,7 +144,9 @@ public Lesson createEditedLesson(Lesson lessonToEdit, EditLessonDescriptor editL
             throws CommandException {
         assert lessonToEdit != null;
         Date updatedDate = editLessonDescriptor.getDate().orElse(lessonToEdit.getStartDate());
-        Date updatedEndDate = editLessonDescriptor.getEndDate().orElse(lessonToEdit.getEndDate());
+        Date updatedEndDate = lessonToEdit.isRecurring()
+                ? editLessonDescriptor.getEndDate().orElse(lessonToEdit.getEndDate())
+                : updatedDate;
```
Comment (eeliana) [inline src/main/java/seedu/address/logic/commands/LessonEditCommand.java:149]: Code for bug fix

Comment (eeliana) [reply src/main/java/seedu/address/logic/commands/LessonEditCommand.java:149]: Rest is adding clashing lessons message 

Tests are for ladd, ledit with end date

File: src/main/java/seedu/address/model/lesson/Lesson.java [hunk 2]
```diff
@@ -224,6 +224,23 @@ public int hashCode() {
                 lessonRates, outstandingFees, cancelledDates);
     }
 
+    public String getLessonDetails() {
```
Comment (lingshanng) [inline src/main/java/seedu/address/model/lesson/Lesson.java:227]: add javadocs

**Review by lingshanng — APPROVED** [review]
LGTM

**Pull request merged**